pkg/runtime/runtime-manifest.go

```diff
@@ -424,43 +434,246 @@ func (m *RuntimeManifest) ResolveStep(step *manifest.Step) error {
 	return nil
 }
 
-// Prepare prepares the runtime environment prior to step execution
-func (m *RuntimeManifest) Prepare() error {
+// Init prepares the runtime environment prior to step execution
+func (m *RuntimeManifest) Initialize() error {
+	if err := m.createOutputsDir(); err != nil {
+		return err
+	}
+
 	// For parameters of type "file", we may need to decode files on the filesystem
 	// before execution of the step/action
-	for _, param := range m.Parameters {
-		// Update ApplyTo per parameter definition and manifest
-		param.UpdateApplyTo(m.Manifest)
+	for paramName, param := range m.bundle.Parameters {
+		if !param.AppliesTo(m.Action) {
+			continue
+		}
 
-		if !param.AppliesTo(string(m.Action)) {
+		def, hasDef := m.bundle.Definitions[param.Definition]
+		if !hasDef {
 			continue
 		}
 
-		if param.Type == "file" {
+		if m.bundle.IsFileType(def) {
 			if param.Destination.Path == "" {
-				return fmt.Errorf("destination path is not supplied for parameter %s", param.Name)
+				return fmt.Errorf("destination path is not supplied for parameter %s", paramName)
 			}
 
 			// Porter by default places parameter value into file determined by Destination.Path
 			bytes, err := m.FileSystem.ReadFile(param.Destination.Path)
 			if err != nil {
-				return fmt.Errorf("unable to acquire value for parameter %s", param.Name)
+				return errors.Wrapf(err, "unable to acquire value for parameter %s", paramName)
 			}
 
+			// TODO(carolynvs): hack around parameters ALWAYS being injected even when empty files mess things up
```

Comment:
I logged a bug against cnab-go https://github.com/cnabio/cnab-go/issues/266 and will work on fixing this after I get this stuff in.